shared/common-adapters/search-filter.js

```diff
@@ -0,0 +1,317 @@
+// @flow
+import * as React from 'react'
+import Box, {Box2} from './box'
+import ClickableBox from './clickable-box'
+import NewInput from './new-input'
+import PlainInput from './plain-input'
+import Text from './text'
+import ProgressIndicator from './progress-indicator'
+import Icon, {type IconType} from './icon'
+import * as Styles from '../styles'
+import * as Platforms from '../constants/platform'
+
+const KeyHandler = (Styles.isMobile ? c => c : require('../util/key-handler.desktop').default)(() => null)
+
+const Kb = {
+  Box,
+  Box2,
+  ClickableBox,
+  Icon,
+  NewInput,
+  PlainInput,
+  ProgressIndicator,
+  Text,
+}
+
+type Props = {|
+  icon?: ?IconType,
+  focusOnMount?: ?boolean,
```

Comment:
You don't have to make everything `foo?: ?bar`. That's for the specific case where: you can optionally supply foo, and if you do you can even pass in null/undefined.

I think what you want for most of these is: 
```foo?: bar```
You can optional supply `foo`, but if you do, then you **must** supply a `bar`

Reply:
I was under the impression that it'd be useful in the situation where we have 

```
type PropsA = {
  foo?: ?bar,
}

type PropsB = {
  foo?: ?bar,
}
```

Then when component B renders a component A, it can just do `<A foo={props.foo />`. If it were `foo?: bar`, then it can't right?

Reply:
I'm actually not sure about that, I'd have to try it out.